dpnp/fft/dpnp_utils_fft.py

```diff
@@ -0,0 +1,212 @@
+        local_shape = (-1,) + a.shape[-len_axes:]
+        a = dpnp.reshape(a, local_shape)
+        index = 1
+
+    shape = a.shape[index:]
+    strides = (0,) + a.strides[index:]
+
+    if a.dtype == dpnp.complex64:
+        dsc = fi.Complex64Descriptor(shape)
+    else:
+        dsc = fi.Complex128Descriptor(shape)
+
+    dsc.fwd_strides = strides
+    dsc.bwd_strides = dsc.fwd_strides
+    dsc.transform_in_place = False
+    if axes is not None:  # batch_fft
+        dsc.fwd_distance = a.strides[0]
+        dsc.bwd_distance = dsc.fwd_distance
+        dsc.number_of_transforms = numpy.prod(a.shape[0])
+    dsc.commit(a.sycl_queue)
+
+    res = dpt.usm_ndarray(
+        a.shape,
+        dtype=a.dtype,
+        buffer=a.usm_type,
+        strides=a.strides,
+        offset=0,
+        buffer_ctor_kwargs={"queue": a.sycl_queue},
+    )
+    fft_event, _ = fi.compute_fft(dsc, a.get_array(), res, is_forward, dev_list)
+    hev_list.append(fft_event)
+    dpctl.SyclEvent.wait_for(hev_list)
+
+    res = dpnp_array._create_from_usm_ndarray(res)
+
+    scale = numpy.prod(shape, dtype=a.real.dtype)
+    norm_factor = 1
+    if norm == "ortho":
+        norm_factor = numpy.sqrt(scale)
+    elif norm == "forward" and is_forward:
+        norm_factor = scale
+    elif norm in [None, "backward"] and not is_forward:
+        norm_factor = scale
+
+    res /= norm_factor
+    if axes is not None:  # batch_fft
+        res = dpnp.reshape(res, a_shape_orig)
+        res = dpnp.moveaxis(res, local_axes, axes)
+    return res
+
+
+def _truncate_or_pad(a, shape, axes):
+    """Truncating or zero-padding the input array along the specified axes."""
+
+    copy_ht_ev = []
+    copy_dp_ev = []
+    shape = (shape,) if isinstance(shape, int) else shape
+    axes = (axes,) if isinstance(axes, int) else axes
+
+    for s, axis in zip(shape, axes):
+        a_shape = list(a.shape)
+        index = [slice(None)] * a.ndim
+        if s == a_shape[axis]:
+            pass
+        elif s < a_shape[axis]:
+            # truncating
+            index[axis] = slice(0, s)
+            a = a[tuple(index)]
+        else:
+            # zero-padding
+            exec_q = a.sycl_queue
+            index[axis] = slice(0, a_shape[axis])  # orig shape
+            a_shape[axis] = s  # modified shape
+            z = dpnp.zeros(
+                a_shape,
+                dtype=a.dtype,
+                usm_type=a.usm_type,
+                sycl_queue=exec_q,
+            )
+            ht_ev, dp_ev = ti._copy_usm_ndarray_into_usm_ndarray(
+                src=a.get_array(),
+                dst=z.get_array()[tuple(index)],
+                sycl_queue=exec_q,
+            )
+            copy_ht_ev.append(ht_ev)
+            copy_dp_ev.append(dp_ev)
+            a = z
+
+    return a, copy_ht_ev, copy_dp_ev
+
+
+def dpnp_fft(a, is_forward, n=None, axis=-1, norm=None):
+    """Calculates 1-D FFT of the input array along axis"""
+
+    dpnp.check_supported_arrays_type(a)
```

Comment:
Most common we use `dpnp.check_supported_arrays_type` directly in interface file.